controllers/datafoundationclient.go

```diff
@@ -124,6 +132,29 @@ func (r *dataFoundationClientReconciler) reconcilePhases() (ctrl.Result, error)
 	}
 
 	if !r.offering.DeletionTimestamp.IsZero() {
+		if pvCount, err := r.countOCSVolumes(); err != nil {
+			return ctrl.Result{}, err
+		} else if pvCount > 0 {
+			r.Log.Info("PVs provisioned by ODF storage classes exists, cannot proceed with uninstallation", "Number of PVs", pvCount)
+			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+
+		storageClassClaimList := &ocsclient.StorageClassClaimList{}
+		if err := r.unrestrictedList(storageClassClaimList); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to list storage class claims: %v", err)
+		}
+		for i := range storageClassClaimList.Items {
+			storageClassClaim := &storageClassClaimList.Items[i]
+			r.Log.Info("Issuing a delete for storageClassClaim %s", storageClassClaim.Name)
+			if err := r.unrestrictedDelete(storageClassClaim); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to issue a delete for storageClassClaim %s: %v", storageClassClaim.Name, err)
+			}
+		}
```

Comment:
Two issues I see with this loop.
1. if one SCC delete fails, you will not issue and delete for anything else after it.
2. You might be issuing a delete for an already "marked for deletion" SCC again and again and again.

Reply:
updated